File: С#/lab1/Vector.cs
using System;

class Vector
{
    private double x, y; 

    public Vector(double x = 0, double y = 0)
    {
        this.x = x;
        this.y = y;
    }

    public double X
    {
        get { return x; }
        set { x = value; }
    }

    public double Y
    {
        get { return y; }
        set { y = value; }
    }

    public double GetMagnitude()
    {
        return Math.Sqrt(x * x + y * y);
    }

    public double GetAngle()
    {
        return Math.Atan2(y, x);
    }
}

File: src/labs/first/Vector.java
package labs.first;

public class Vector {
    private double x;
    private double y;

    public Vector() {
        this.x = 0.0;
        this.y = 0.0;
    }

    public Vector (double x, double y) {
        this.x = x;
        this.y = y;
    }

    public Vector (Vector v) {
        this.x = v.x;
        this.y = v.y;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public void setX(double x) {
        this.x = x;
    }
    public void setY(double y) {
        this.y = y;
    }

    public double getRadius(){
        return Math.sqrt(x * x + y * y);
    }

    public double getAngle() {
        return Math.atan2(y, x);
    }

    @Override
    public String toString() {
        return "Vector (" + x + ", " + y + ")";
    }

    @Override
    protected void finalize() throws Throwable {
        System.out.println("Vector" + this + " is being destroyed");
    }
}
